### ISSUE TITLE
`GenericMetricLayer::pair` doesn't allow the option to manually initialize the recorder

### ISSUE THREAD

**Issue by Ptrskay3** [Ptrskay3/axum-prometheus#48]
Source: https://github.com/Ptrskay3/axum-prometheus/discussions/47.

The `GenericMetricLayer::pair` API doesn't really give any options to initialize the struct manually.  We'd like to write the following:
```rust
struct Recorder<'a> {
    host: String,
    prefix: Option<&'a str>
}

impl<'a> MakeDefaultHandle for Recorder<'a> {
    type Out = ();

    fn make_default_handle() -> Self::Out {
        let recorder = StatsdBuilder::from(self.host, 8125)
            .build(self.prefix)
            .expect("Could not create StatsdRecorder");

        metrics::set_global_recorder(recorder).unwrap();
    }
}
```
Currently we can't initialize the struct so that `axum_prometheus` uses it, neither can we use `self.host` in the trait in any meaningful way.

**Comment by elrob** [comment]
Yeah, this example with the `Recorder` with configuration values was exactly the direction I was going in until I realized it didn't seem possible.

**Comment by Ptrskay3** [comment]
Sorry for the super late reaction. Just merged support for this in #49! I'll have to look at some other issues as well, but this will be released soon.
